File: Components/ASP.NET/VB/ASPxSpreadsheetDemos/App_Code/DemoHelpers/ExpensesModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Xml.Serialization

Namespace DevExpress.Web.Demos
	<XmlRoot("Expenses")> _
	Public Class ExpensesReport
		Public Sub New()
			Info = New ExpensesInfo()
			Expenses = New List(Of Expense)()
		End Sub

		Public Property Info() As ExpensesInfo
		Public Property Expenses() As List(Of Expense)
	End Class

	Public Class ExpensesInfo
		Public Property Purpose() As String
		Public Property StatementNumber() As Integer
		Public Property EmployeeID() As Integer
		Public Property EmployeeName() As String
		Public Property EmployeePosition() As String
		Public Property SSN() As String
		Public Property EmployeeDepartment() As String
		Public Property Manager() As String
	End Class

	Public Class Expense
		Public Property CreationDate() As DateTime
		Public Property Account() As Integer
		Public Property Description() As String
		Public Property Hotel() As Double
		Public Property Transport() As Double
		Public Property Fuel() As Double
		Public Property Meals() As Double
		Public Property Phone() As Double
		Public Property Misc() As Double
		Public Property Entertainment() As Double
	End Class
End Namespace

File: Components/ASP.NET/CS/MVCxSpreadsheetDemos/Models/ExpensesModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Xml.Serialization;

namespace DevExpress.Web.Demos {
    [XmlRoot("Expenses")]
    public class ExpensesReport {
        public ExpensesReport() {
            Info = new ExpensesInfo();
            Expenses = new List<Expense>();
        }

        [XmlElement]
        public ExpensesInfo Info { get; set; }
        [XmlElement("Expense")]
        public List<Expense> Expenses { get; set; }
    }

    public class ExpensesInfo {
        [XmlAttribute]
        public string Purpose { get; set; }
        [XmlAttribute]
        public int StatementNumber { get; set; }
        [XmlAttribute]
        public int EmployeeID { get; set; }
        [XmlAttribute]
        public string EmployeeName { get; set; }
        [XmlAttribute]
        public string EmployeePosition { get; set; }
        [XmlAttribute]
        public string SSN { get; set; }
        [XmlAttribute]
        public string EmployeeDepartment { get; set; }
        [XmlAttribute]
        public string Manager { get; set; }
    }

    public class Expense {
        [XmlAttribute(AttributeName = "Date")]
        public DateTime CreationDate { get; set; }
        [XmlAttribute]
        public int Account { get; set; }
        [XmlAttribute]
        public string Description { get; set; }
        [XmlAttribute]
        public double Hotel { get; set; }
        [XmlAttribute]
        public double Transport { get; set; }
        [XmlAttribute]
        public double Fuel { get; set; }
        [XmlAttribute]
        public double Meals { get; set; }
        [XmlAttribute]
        public double Phone { get; set; }
        [XmlAttribute]
        public double Misc { get; set; }
        [XmlAttribute]
        public double Entertainment { get; set; }
    }
}
